packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,157 @@
+# Table Column Widths
+
+## Problem Statement
+
+We need to provide users the means for changing the widths of individual columns as needed. This should be able to be accomplished both through interactive means as well as via a programmatic API.
+
+## Links To Relevant Work Items and Reference Material
+
+- [#873 Programmatically resize column width](https://github.com/ni/nimble/issues/873)
+- [#846 Interactively resize column width](https://github.com/ni/nimble/issues/846) 
+- [Table Design Doc](https://xd.adobe.com/view/5b476816-dad1-4671-b20a-efe796631c72-0e14/screen/d389dc1e-da4f-4a63-957b-f8b3cc9591b4/specs/)
+
+
+## Implementation / Design
+
+### Expected Behavior
+
+The behavior that has been prescribed for column sizing is as follows:
+
+- Columns should be able to be configured to either maintain a fixed width, or grow proportionally with the table such as when the window resizes causing the table width to increase. Tables can consist of columns that are configured as a mixture of the two modes.
+- If a user drags a divider between two columns to the right, then the column on the left will grow larger, and the column on the right will grow smaller by the same pixel amount. Sub-behaviors to this are:
+    - If a shrinking column has reached its minimum pixel size, then the next column in the direction of the sizing action will be affected up to the final column in a given direction.
+    - A sizing action to the left will ultimately stop having an effect when the left-most column reaches its minimum size.
+    - A sizing action to the right that would ultimately result in the final right column reaching its minimum size (all columns still within current table width) would then begin to push columns out of the table width resulting in a horizontal scrollbar.
+- Columns can be configured to not allow a user to interactively size them
+    - The implicit behavior present based on the behaviors described above, is that in a sizing action that cascades to a column configured to not be resized is that the column won't be resized towards a minimum size, and the cascade will effectively "skip" this column.
+
+### API
+
+To accomodate the various sizing modes of a column, in addition to the other necessary behaviors, we can add the following properties to `TableColumn`:
+
+```
+class TableColumn {
+    // when set 'currentFractionalWidth' should be ignored
+    @observable
+    public currentFixedWidth: number | null = null;
```

Comment:
Should we consider calling this `currentPixelWidth` instead of `currentFixedWidth`? Possible rationale:
1. If we were to allow non-fractional columns to be resized, then "fixed" seems a bit confusing because the size can change.
2. It encodes the units in the name, reducing the odds that someone might think they could use CSS length units like `em` 

I admit those are both weak rationale and if we stick with making these non-resizable then I think `currentFixedWidth` is probably ok, but want to open it for discussion.

Reply:
I'm happier with `currentPixelWidth`.